### PR TITLE
switch to dune

### PR DESCRIPTION
**Pull request by redianthus**
Hi,

It's still not building but it uses *not building while using dune*, which is better.

I had to do a few things apart from renaming and removing the parenthesis, but the diff is not showing them clearly. It's mainly related to dune not having the `$<` or `$@` anymore.

Cheers !

### CODE REVIEW

File: config/jbuild [hunk 1]
```diff
@@ -1,22 +0,0 @@
...
-            wayland-server-ccopt.sexp
-            wayland-server-cclib.sexp
-            wayland-server-ccopt
-            wayland-server-cclib
-            pixman-1-ccopt.sexp
-            pixman-1-cclib.sexp
-            pixman-1-ccopt
-            pixman-1-cclib))
-  (deps    (discover.exe))
-  (action  (run ${<} -ocamlc ${OCAMLC}))))
```
Comment (Armael) [inline config/jbuild:20]: What happens to the `-ocamlc ${OCAMLC}` part? Was it useless?

Comment (redianthus) [reply config/jbuild:20]: If I understand correctly, it's now handled by dune:

> no need to manually pass -ocamlc flag

See [dune doc](https://dune.readthedocs.io/en/latest/configurator.html?highlight=ocamlc).

Comment (Armael) [reply config/jbuild:20]: :+1:

**Comment by Armael** [comment]
Thanks!

**Pull request merged**